documentation/solution-architecture.md

```diff
@@ -0,0 +1,151 @@
+
+This document describes a high level solution architecture with software components as most detailed level and following ISO-42010 and NORA five layer model in a most pragmatic lean way intended to be readable for all stakeholders involved.
+
+Note: expect this document to change as it is work in progress.
+
+## Foundational layer
+
+The solution might need to be compliant with the following laws, architectures and regulations:
+
+- NORA / GEMMA, and all implications from this though noteworthy:
+- EIF/EiDAS
+- GDPR
+- EN 301 549 / WCAG 2.0 (accessibility)
+- ETSI TS 103 171 (XAdES) for paper wallet signatures
+- BIR2017 (security management)
+- REGULATION (EU) No 910/2014 (as far as applicable)
+
+Note that the core waardepapieren solution itself does not implement or include identification / authentication services for citizens
+nor any key distribution mechanism which is more dependent on it's integration and the specific platforms being used.
+
+As for NORA and GEMMA, Discipl aims at a new innovative architecture to be integrated in NORA in the Future, aiming at a society that is facilitated with a hybrid distributed linked data landscape that aligns with concepts like attribute based credentials and self sovereign identity. And something that is called "Common Ground" is a similair innovation program for GEMMA. This is why we take the liberty to aim the solution to adhere to the principles of Discipl and common ground over those articulated in NORA and GEMMA and defer the identification and resolution of possible discrepancies within these principles to the ongoing development of discipl, common ground (including NLX).
+
+As NLX aims at conforming to EIF/EiDAS and by having the waardepapieren solution use this platform for information retrieval and the NLX certificates for signing (and validation) and by having it not include the identification / authentication of actors, this solution quite automaticly adheres to EIF/EiDAS.
+
+As already mentioned, the solution does not store any personal information on ledgers. All information retrieved through NLX are only held temporarily in memory to be able to issue the information to the holder of the requested credential, often being the subject of the information. Though in practice the holders also can be actors that are allowed to retrieve proofs for subjects other then themselves, for instance notaries, we focus on the case where the holder is the subject itself and defer other types of holders to the integration projects of the waardepapieren solution. The reason for this is that Discipl can easily support these alternative authorizations. By doing so, as long as the issuer and validating actor complies to GDPR in relation to the information subject, so does this solution. The waardepapieren solution helps complying to GDPR by holding information only temporarily and in encrypted state. The software is open source and is to be independently audited on the topic of security.
```

Comment:
"in encrypted state" not true at the moment, if it's a must-have we probably should have an issue for it.

Reply:
Ok changed it for now.